### ISSUE TITLE
it doesn't work on mac2017pro 13'' with dell p2415q

### ISSUE THREAD

**Issue by kevin-lucifer** [Floris497/mac-pixel-clock-patch-V2#278]
mac-pixel-clock-patch-V2 git:(master) ./CoreDisplay-patcher.command status
SIP looks to be disabled, all good!

(otool) Detected patched CoreDisplay on OS X 10.13.4 (17E199).

System Report -> Graphics/Displays it's @30Hz

**Comment by Floris497** [comment]
Yep, you have the wrong cable..

**Comment by kevin-lucifer** [comment]
type-c to hdmi ,10gps cable , is this cable okay for 4k monitor@60hz? I'm confused by the standard.

**Comment by kevin-lucifer** [comment]
Here is the patch log, I got some warning, FYI.

SIP looks to be disabled, all good!

(otool) Detected unpatched CoreDisplay on OS X 10.13.4 (17E199).
SIP looks to be disabled, all good!
Patching CoreDisplay with patch version 4
Password:
Re-signing /System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay
/System/Library/Frameworks/CoreDisplay.framework/Versions/A/CoreDisplay: replacing existing signature
Running 'sudo update_dyld_shared_cache' (Might need re-run after reboot) - CAN TAKE A WHILE (Maybe even 10/30 minutes)update_dyld_shared_cache: warning @rpath install name for system framework: /System/Library/PrivateFrameworks/AssetCacheServicesExtensions.framework/Versions/A/XPCServices/AssetCacheTetheratorService.xpc/Contents/Frameworks/MobileDeviceKitLite.framework/Versions/A/MobileDeviceKitLite
update_dyld_shared_cache: warning @rpath install name for system framework: /System/Library/PrivateFrameworks/BiometricKitUI.framework/Versions/A/BiometricKitUI
update_dyld_shared_cache: warning @rpath install name for system framework: /System/Library/PrivateFrameworks/BookKit.framework/Versions/A/Frameworks/BookKitFoundation.framework/Versions/A/BookKitFoundation
update_dyld_shared_cache: warning @rpath install name for system framework: /System/Library/PrivateFrameworks/BroadcomWLAN.framework/Versions/A/BroadcomWLAN
update_dyld_shared_cache: warning @rpath install name for system framework: /System/Library/PrivateFrameworks/SetupAssistantSupport.framework/Versions/A/SetupAssistantSupport
update_dyld_shared_cache: warning @rpath install name for system framework: /System/Library/PrivateFrameworks/Swift/libswiftDemangle.dylib
update_dyld_shared_cache: warning @rpath install name for system framework: /System/Library/PrivateFrameworks/Swift/libswiftRemoteMirror.dylib
update_dyld_shared_cache: warning @rpath install name for system framework: /System/Library/PrivateFrameworks/WirelessDiagnosticsSupport.framework/Versions/A/WirelessDiagnosticsSupport
update_dyld_shared_cache: i386 incorporating 624 OS dylibs, tracking 1040 others, building closures for 252 executables
update_dyld_shared_cache: x86_64h incorporating 1264 OS dylibs, tracking 1779 others, building closures for 2114 executables
update_dyld_shared_cache: warning: i386 adding dependents to /System/Library/PrivateFrameworks/Assistant.framework/Versions/A/Resources/Assistant Plugin.ibplugin/Contents/MacOS/Assistant Plugin: required dylib '@rpath/InterfaceBuilderKit.framework/Versions/A/InterfaceBuilderKit' not found, needed by '/System/Library/PrivateFrameworks/Assistant.framework/Versions/A/Resources/Assistant Plugin.ibplugin/Contents/MacOS/Assistant Plugin'
update_dyld_shared_cache: warning: x86_64h adding dependents to /System/Library/BridgeSupport/ruby-2.3/bridgesupportparser.bundle: required dylib '@rpath/libclang.dylib' not found, needed by '/System/Library/BridgeSupport/ruby-2.3/bridgesupportparser.bundle'
update_dyld_shared_cache: warning: x86_64h adding dependents to /System/Library/PrivateFrameworks/Assistant.framework/Versions/A/Resources/Assistant Plugin.ibplugin/Contents/MacOS/Assistant Plugin: required dylib '@rpath/InterfaceBuilderKit.framework/Versions/A/InterfaceBuilderKit' not found, needed by '/System/Library/PrivateFrameworks/Assistant.framework/Versions/A/Resources/Assistant Plugin.ibplugin/Contents/MacOS/Assistant Plugin'
update_dyld_shared_cache: warning: x86_64h adding dependents to /System/Library/QuickLook/Audio.qlgenerator/Contents/MacOS/Audio: required dylib '/Library/Frameworks/iTunesLibrary.framework/Versions/A/iTunesLibrary' not found, needed by '/System/Library/QuickLook/Audio.qlgenerator/Contents/MacOS/Audio'
Closing..

**Comment by kevin-lucifer** [comment]
And here is my hardware info:

  Model Name:	MacBook Pro
  Model Identifier:	MacBookPro14,1
  Processor Name:	Intel Core i7
  Processor Speed:	2.5 GHz
  Number of Processors:	1
  Total Number of Cores:	2
  L2 Cache (per Core):	256 KB
  L3 Cache:	4 MB
  Memory:	16 GB
  Boot ROM Version:	MBP141.0173.B00
  SMC Version (system):	2.43f6

**Comment by Floris497** [comment]
What exact cable? It's better to have DisplayPort.. most USB-C to HDMI don't support HDMI2.0 (it should specifically say HDMI2.0) but on the other hand MacOS kinda sucks with support for HDMI2.0 anyways...

**Comment by Floris497** [comment]
But u definitely don't need the patch on your machine :)

**Comment by kevin-lucifer** [comment]
here is my system info after patch, I got a  UI Looks like:	1920 x 1080 @ 30 Hz

  Displays:
SwitchResX4 - Color LCD:
  Display Type:	Built-In Retina LCD
  Resolution:	2560 x 1600 Retina
  Framebuffer Depth:	24-Bit Color (ARGB8888)
  Main Display:	Yes
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Automatically Adjust Brightness:	No
SwitchResX4 - DELL P2415Q:
  Resolution:	3840 x 2160 (2160p 4K UHD - Ultra High Definition)
  UI Looks like:	1920 x 1080 @ 30 Hz
  Framebuffer Depth:	30-Bit Color (ARGB2101010)
  Display Serial Number:	3VV1475I0YLL
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Automatically Adjust Brightness:	No
  Connection Type:	DVI or HDMI
  Adapter Firmware Version:	 f.1d

**Comment by Floris497** [comment]
Yep, but talking about the cable, not the mac ;) also Dell is kinda shitty supported for mac on 4K.. can u send me link to specific cabel u have?

**Comment by kevin-lucifer** [comment]
I have no link about my cable, maybe I should buy one support hdmi2.0 myself. Btw,I find out some some type-c to dp adapter on amz , are they pure digital signal, or a->d , d->a signal transfer in the adapter?

**Comment by kevin-lucifer** [comment]
Finally I decide to buy a type-c to dp cable. The best choice to save time and money. lol, thanks Floris.

**Comment by Floris497** [comment]
Yep, pure digital, if you want to wait a month there is pretty good options on AliExpr also

**Comment by kevin-lucifer** [comment]
I bought a type-c to mini dp adapter, only @30hz refresh rate. Any recommend cable?

**Comment by kevin-lucifer** [comment]
Finally, @60hz lol.

  Displays:
SwitchResX4 - Color LCD:
  Display Type:	Built-In Retina LCD
  Resolution:	2560 x 1600 Retina
  Framebuffer Depth:	24-Bit Color (ARGB8888)
  Main Display:	Yes
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Automatically Adjust Brightness:	No
SwitchResX4 - DELL P2415Q:
  Resolution:	3840 x 2160 (2160p 4K UHD - Ultra High Definition)
  UI Looks like:	1920 x 1080 @ 60 Hz
  Framebuffer Depth:	30-Bit Color (ARGB2101010)
  Display Serial Number:	3VV1475I0YLL
  Mirror:	Off
  Online:	Yes
  Rotation:	Supported
  Adapter Type:	DisplayPort
  Automatically Adjust Brightness:	No
